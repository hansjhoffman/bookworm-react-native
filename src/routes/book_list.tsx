import React from "react";
import { FlatList, StyleSheet, View } from "react-native";
import { NavigationScreenProp } from "react-navigation";
import gql from "graphql-tag";
import { useQuery } from "@apollo/react-hooks";

import { Card, Error, Loading } from "../components";
import { Colors, Layout } from "../constants";
import { Book } from "../types";

export const BOOKS_QUERY = gql`
  query bookList {
    books {
      id
      authors
      rating
      thumbnail
      title
    }
  }
`;

type Props = {
  navigation: NavigationScreenProp<{}>;
};

export function BookList({ navigation }: Props) {
  const { data, error, loading } = useQuery(BOOKS_QUERY);

  if (loading) return <Loading />;
  if (error) return <Error message={error.message} />;

  function _keyExtractor(item: Book): string {
    return item.id;
  }

  function _renderItem({ item }: { item: Book }) {
    return (
      <Card
        key={item.id}
        book={item}
        onPress={() =>
          navigation &&
          navigation.navigate({
            routeName: "book_detail",
            params: { id: item.id },
          })
        }
      />
    );
  }

  return (
    <View style={styles.container}>
      <FlatList
        contentContainerStyle={styles.columnWrapper}
        columnWrapperStyle={styles.multiColumns}
        data={data.books}
        initialNumToRender={6}
        keyExtractor={_keyExtractor}
        numColumns={2}
        renderItem={_renderItem}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  columnWrapper: {
    paddingTop: 100,
  },
  centered: {
    alignItems: "center",
    justifyContent: "center",
    padding: Layout.margin_lg,
  },
  container: {
    backgroundColor: Colors.white,
    flex: 1,
  },
  menuIcon: {
    height: 23,
    width: 23,
    marginLeft: Layout.margin_lg,
    resizeMode: "contain",
    tintColor: Colors.iconSelected,
  },
  multiColumns: {
    justifyContent: "space-evenly",
  },
});
