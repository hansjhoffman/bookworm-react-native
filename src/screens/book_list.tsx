import React from "react";
import { Text, View } from "react-native";
import { NavigationContainerProps } from "react-navigation";

export function BookList({ navigation }: NavigationContainerProps) {
  return (
    <View>
      <Text
        onPress={() =>
          navigation &&
          navigation.navigate({
            routeName: "book_detail",
            params: { id: 1 },
          })
        }>
        Book 1
      </Text>
      <Text
        onPress={() =>
          navigation &&
          navigation.navigate({
            routeName: "book_detail",
            params: { id: 2 },
          })
        }>
        Book 2
      </Text>
    </View>
  );
}
