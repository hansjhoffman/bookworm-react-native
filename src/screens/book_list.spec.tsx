import React from "react";
import { render } from "@testing-library/react-native";

import { BookList } from "./book_list";

describe("[Screen] book_list", function() {
  it("should match snapshot", function() {
    const { baseElement } = render(<BookList />);

    expect(baseElement).toMatchSnapshot();
  });
});
