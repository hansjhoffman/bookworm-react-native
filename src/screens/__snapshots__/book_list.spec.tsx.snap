// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Screen] book_list should match snapshot 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <Text>
        Book 1
      </Text>
      <Text>
        Book 2
      </Text>
    </View>
  </View>
</View>
`;
